frontend/api/petApi.ts

```diff
@@ -1,65 +1,72 @@
-import { Shelter, Animal } from "@contract";
+import { Shelter, Animal, Upload } from "@contract";
+import { NewAnimal } from "common/types";
 import { BaseApi } from "./baseApi";
 
 export class PetApi extends BaseApi {
-	public async fetchShelterPets(
-		shelterId: string,
-	): Promise<Shelter.fetchPetsApiDomain.response> {
-		// const query: Shelter.fetchPetsApiDomain.requestQuery = {
-		// 	shelterId,
-		// };
-		// const res = await this.fetch(
-		// 	Shelter.fetchPetsApiDomain.method,
-		// 	Shelter.fetchPetsApiDomain.endpoint,
-		// 	query,
-		// );
-		await delay();
-		return generateMockPetData();
+	public async fetchShelterPets(shelterId: string) {
+		const query: Shelter.fetchPetsApiDomain.requestQuery = {
+			shelterId,
+		};
+		const res = await this.request<Shelter.fetchPetsApiDomain.response>(
+			Shelter.fetchPetsApiDomain.method,
+			Shelter.fetchPetsApiDomain.endpoint,
+			query,
+		);
+
+		return res!.data;
```

Comment:
When will there not be data? I believe there would a non 2xx status if no data?

Reply:
this is more for typescript where it thinks res might be undefined - it's axios' response typing i think

Reply:
This is actually from a switch with no default value in

```
	protected fetch<Params, Response>(
		method: Methods,
		url: string,
		params?: Params,
	) {
		switch (method) {
			case Methods.GET:
				return this.get<Params, Response>(url, params);
			case Methods.POST:
				return this.post<Params, Response>(url, params);
		}
	}
```

Changing the second line to just include the handled methods seems to resolve this.
```
method: Methods.GET | Methods.POST
```